src/tsserver/server.ts

```diff
@@ -484,8 +484,9 @@ namespace ts.server {
         private constructed: boolean | undefined;
 
         constructor() {
+            let isConstructed = false;

```

Comment:
Instead of this I would move the logic of scheduling things into method event itself